tests/cli/test_rasa_data.py

```diff
@@ -349,6 +369,7 @@ def test_validate_files_exit_early():
 def test_validate_files_invalid_domain():
     args = {
         "domain": "data/test_domains/default_with_mapping.yml",
+        "fail_on_warnings": False,
```

Comment:
I don't remember this being required on the other branch 🤔 Why so?
Also the `test_rasa_train` wasn't updated, is that because the validation improvement to rasa train was released in 3.6.0 only?